### PR TITLE
Updated contribution guidelines

### PR DESCRIPTION
**Pull request by nfode**
I updated the contribution guidelines according to #114.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,26 +30,57 @@ Our project uses checkstyle to ensure coding standards. If you want to read more
...
+5. `git reset origin/master` - this prepares that all commits can be squashed togheter: The local checkout (“working tree”) is left untouched, but the “pointer” of the current branch is reset to `origin/master`.
+6. Check changes in your favourite git tool:
+	- Is each of your changes recognized?
+	- Are there to much changed files? - Do not stage things you did not intend to change.
+	- Check if your Code follows our style guidelines.
+7. `git add .` - stage your changes for the commit.
+8. `git commit` - commit your changes with a meaningfull title and description.
+9. Force push your changes with `git push -f` to overwrite the remote commits.
+
+You dont habe to use the `git reset` - Method you also can use `git rebase -i <commit-id>` (see [stackoverflow](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git)) to squash your commits. But in general the result should be the same.
```
Comment (c-mueller) [inline CONTRIBUTING.md]: Little typo

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,26 +30,57 @@ Our project uses checkstyle to ensure coding standards. If you want to read more
...
+To get a review remove the [WIP] tag and assign reviewers. If they submitted a review react to their changes and update your feature. A Pull Request is ready to merge if the reviewers approved it.
 
-### Merging a Pull Request
+### Prepare a Pull Request
 Before a pull request can be merged, it must fulfill the criteria specified in the [Definition of Done](/docs/dev/dod.md).
-Merging should be done through the **squash and merge** option in GitHub. This allows all to be combined into one commit on the master branch in order to not clutter up the commit history.
+
+The aim of the following steps is to create a **single commit**.
+
+Steps to prepare the pull request (reference [Winery - Prepare a Pull Request](https://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request)):
```
Comment (c-mueller) [inline CONTRIBUTING.md]: Keep in mind that `origin` is used for branches within the same repository and `upstream` is used if the branch that should get merged comes from a fork.

I would add a note to this point because we might (just in case) have contributors that submit a pull request from a fork

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -30,26 +30,57 @@ Our project uses checkstyle to ensure coding standards. If you want to read more
...
-When a feature is considered finished it should be **reviewed** before merging the corresponding PR. During this stage the [WIP] label in the title should be removed.
+When a feature is considered finished it is necessary to get a **review**.
+To get a review remove the [WIP] tag and assign reviewers. If they submitted a review react to their changes and update your feature. A Pull Request is ready to merge if the reviewers approved it.
 
-### Merging a Pull Request
+### Prepare a Pull Request
 Before a pull request can be merged, it must fulfill the criteria specified in the [Definition of Done](/docs/dev/dod.md).
-Merging should be done through the **squash and merge** option in GitHub. This allows all to be combined into one commit on the master branch in order to not clutter up the commit history.
+
+The aim of the following steps is to create a **single commit**.
```
Comment (c-mueller) [inline CONTRIBUTING.md]: Maybe reword this to 
> The goal of the following steps is to get a single commit, containing all differences between the `master` branch and the branch of the pull request.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -30,26 +30,57 @@ Our project uses checkstyle to ensure coding standards. If you want to read more
...
 ## Pull Requests
 Any major changes to existing or the addition of new features or artifacts should be done through pull requests or PRs.
 
 ### Working on a Pull Request
 PRs with unfinished features should have a [WIP] tag at the beginning of their title. This shows that this feature is still in progress and not ready to be merged.
 
 ### Reviewing a Pull Request
-When a feature is considered finished it should be **reviewed** before merging the corresponding PR. During this stage the [WIP] label in the title should be removed.
+When a feature is considered finished it is necessary to get a **review**.
+To get a review remove the [WIP] tag and assign reviewers. If they submitted a review react to their changes and update your feature. A Pull Request is ready to merge if the reviewers approved it.
```
Comment (c-mueller) [inline CONTRIBUTING.md]: Maybe also mention the `Review Q/A` pipeline in ZenHub?

File: CONTRIBUTING.md (hunk 2 above)
Comment (nfode) [reply CONTRIBUTING.md]: So add a general note like this:
> If you forked the repository you have to replace origin with upstream.

**Comment by nfode** [comment]
> Probably fitts our current needs.
Why probably?

That means we can get higher coverage if we add md documents?
